openedx/features/colaraz_features/api/serializers.py

```diff
@@ -0,0 +1,59 @@
+"""
+Serializers for colaraz API.
+"""
+from django.contrib.sites.models import Site
+
+from rest_framework import serializers
+from organizations.models import Organization
+
+from .fields import DomainField
+from ..helpers import create_site, create_site_theme, create_organization, create_site_configuration
+
+
+class SiteOrgSerializer(serializers.Serializer):
+    site_domain = DomainField()
+    site_name = serializers.CharField(max_length=255)
+    site_theme = serializers.CharField(max_length=255)
+    org_name = serializers.CharField(max_length=255)
+    org_short_name = serializers.SlugField(max_length=255)
+    platform_name = serializers.CharField(max_length=255)
+    university_name = serializers.CharField(max_length=255, required=False)
+    organizations = serializers.ListField(
+        child=serializers.SlugField(max_length=255)
+    )
+
+    @staticmethod
+    def validate_site_domain(value):
+        """
+        Validate that site domain does not already exists.
+        """
+        if Site.objects.filter(domain=value).exists():
+            raise serializers.ValidationError('A site with domain "{}" already exists.'.format(value))
+        return value
+
+    @staticmethod
+    def validate_org_name(value):
+        """
+        Validate that site domain does not already exists.
+        """
+        if Organization.objects.filter(name=value).exists():
+            raise serializers.ValidationError('An organization with name "{}" already exists.'.format(value))
+        return value
+
+    def create(self, validated_data):
```

Comment:
We have to create two sites, one for lms and one for studio.
We can do two separate requests to add them separately (but in current scenario that is not handled), or we can do it in one single request as we know what kind of URLs will be made for lms and studio.

Reply:
okay, just to be sure I just need to add `studio` in front of the `site_domain` for studio sight? e.g. if `site_domain` is **example.com** studio will be **studio.example.com**

Reply:
As far as I know, colaraz is going to use `courses.example.com` for `lms` and `studio.example.com` for `studio`

Reply:
Done